src/framework/HadronizationManager.h

```diff
@@ -47,6 +47,15 @@ class HadronizationManager
 		//get Hadrons from Hadronization submodules
   void GetHadrons(vector<shared_ptr<Hadron>>& signal);
 
+  // deletes the hadrons from the different hadronization modules
```

Comment:
The workflow of the Delete functions here is certainly not easy to follow and understand. Since, but I am not 100% sure, these functions are not necessarily used to free up the memory (if so they should be in the Clear() part), maybe one could change Delete to Remove or Select etc since, at least for me, that would indicate an operation not directly related to memory management. But this is not a crucial point.  Not for this PR though, but for XSCAPE 2.0, since SMASH won't be an afterburner per se, would should make sure that we then have a different base class ...